src/vendor/azure-uamqp-c/src/message.c

```diff
@@ -1447,3 +1465,71 @@ int message_get_message_format(MESSAGE_HANDLE message, uint32_t *message_format)
 
     return result;
 }
+
+int message_set_delivery_tag(MESSAGE_HANDLE message, AMQP_VALUE delivery_tag_value)
+{
+    int result;
+
+    if ((message == NULL) || (delivery_tag_value == NULL))
+    {
+        LogError("Bad arguments: message = %p, delivery_tag = %p",
+            message, delivery_tag_value);
+        result = __FAILURE__;
+    }
```

Comment:
Interesting - I don't think this will be an issue here because we're already NULL validating the value before this function, and also this value should never be set twice on a single message (making the reset unnecessary). However I have adopted the same behaviour to be consistent.